xblock/mixins.py

```diff
@@ -516,6 +516,10 @@ def _add_field(self, node, field_name, field):
         as an xml attribute or creates a separate child node.
         """
         value = field.to_string(field.read_from(self))
+
+        if not isinstance(value, basestring):
```

Comment:
Can we be more explicit and check `if value is None` instead? Do we ever expect any other type of value than `basestring` or `None`?